### PR TITLE
add web hook cause

### PR DESCRIPTION
**Pull request by casz**
`Triggered by GitLab Merge Request #1: bob => master`
`Started by GitLab push by Joseph Petersen`

🙌

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebHookCause.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        }
+        return this;
+    }
+
+    public GitLabWebHookCause fromMergeRequest(MergeRequestEvent mergeRequestEvent) {
+        ObjectAttributes objectAttributes = mergeRequestEvent.getObjectAttributes();
+        String id = String.valueOf(objectAttributes.getIid());
+        String sourceNameSpace = objectAttributes.getSource().getNamespace();
+        String targetNameSpace = objectAttributes.getTarget().getNamespace();
+        String namespace =
```
Comment (baymac) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebHookCause.java]: I suggest either you keep `namespace` camel cased or don't.

Comment (casz) [reply src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebHookCause.java]: I am not modifying the user typed name space.
and I am only using it to compare if source and target is the same

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebHookCause.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
```
Comment (casz) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebHookCause.java]:
```suggestion
        String nameSpace =
```

File: src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebHookCause.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    public GitLabWebHookCause fromMergeRequest(MergeRequestEvent mergeRequestEvent) {
+        ObjectAttributes objectAttributes = mergeRequestEvent.getObjectAttributes();
+        String id = String.valueOf(objectAttributes.getIid());
+        String sourceNameSpace = objectAttributes.getSource().getNamespace();
+        String targetNameSpace = objectAttributes.getTarget().getNamespace();
+        String nameSpace =
+            StringUtils.equals(sourceNameSpace, targetNameSpace) ? "" : sourceNameSpace + "/";
+        String source = String.format("%s%s", namespace, objectAttributes.getSourceBranch());
```
Comment (casz) [inline src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabWebHookCause.java]:
```suggestion
        String source = String.format("%s%s", nameSpace, objectAttributes.getSourceBranch());
```

**Pull request merged**